sound/soc/soc-pcm.c

```diff
@@ -1940,7 +1943,8 @@ void dpcm_be_dai_hw_free(struct snd_soc_pcm_runtime *fe, int stream)
 		dev_dbg(be->dev, "ASoC: hw_free BE %s\n",
 			be->dai_link->name);
 
-		__soc_pcm_hw_free(be, be_substream);
+		if (be_substream)
+			__soc_pcm_hw_free(be, be_substream);
```

Comment:
Here checks on "be->dpcm[stream].state"  at L1935 protect against NULL be_substream usage.